src/mrpro/data/IHeader.py

```diff
@@ -92,35 +162,93 @@ def make_unique_tensor(values: Sequence[float]) -> torch.Tensor | None:
             else:
                 return torch.as_tensor(values)
 
-        # Conversion functions for units
-        def ms_to_s(ms: torch.Tensor | None) -> torch.Tensor | None:
-            return None if ms is None else ms / 1000
+        def ensure_unique_string(values: Sequence[str], dicom_tag: TagType) -> str:
+            """Ensure there is only one unique string."""
+            if any(val is None for val in values):
+                return UNKNOWN
+            elif len(set(values)) == 1:
+                return values[0]
+            else:
+                raise ValueError(f'Inconsistent entries for {dicom_tag} found.')
+
+        @overload
+        def ensure_unique_number(values: Sequence[int], dicom_tag: TagType) -> int: ...
+
+        @overload
+        def ensure_unique_number(values: Sequence[float], dicom_tag: TagType) -> float: ...
+
+        def ensure_unique_number(values: Sequence[float | int], dicom_tag: TagType) -> float | int:
```

Comment:
You could use a type var here instead of the overloads

Reply:
from typing import TypeVar

T = TypeVar('T', int, float)

def ensure_unique_number(values: Sequence[T], dicom_tag: TagType) -> T:
    """Ensure there is only one unique number."""
    if len(set(values)) == 1:
        return values[0]
    else:
        raise ValueError(f'Inconsistent entries for {dicom_tag} found.'

Or even just ensure_unique without bounds on T:

T = TypeVar('T')
def ensure_unique(values: Sequence[T], dicom_tag: TagType) -> T:
    """Ensure there is only one unique element"""
    if len(set(values)) == 1:
        return values[0]
    else:
        raise ValueError(f'Inconsistent entries for {dicom_tag} found.'


The typevar line could be removed if we limit ourselves to python 3.12, but is required for 3.11 compatibility.

Reply:
This way,, for a  sequence of Str, the function will return a string, for floats a float, etc.

Reply:
combine string, floats and int and use `if string return UKNOWN else return None`